### DISCUSSION TITLE
Export iTowns Globe to use inside my own ThreeJS instance.

### DISCUSSION THREAD

**Discussion by crubier** [iTowns/itowns#1829] [Ideas]
Currently, iTowns controls the ThreeJS instance (render loop, import, etc.)

I have a use case where I'd like to "reverse the control",  and use iTowns as an object inside of my own ThreeJS instance:

We use react-three-fiber to manage our ThreeJS instance and scene: https://github.com/pmndrs/react-three-fiber , but it cannot integrate with iTowns, since they both need to control the ThreeJS render loop.

So my request would be: could we export an iTowns globe as an "Object" that I can add to my own ThreeJS instance ?

As a reference, The NYTimes released a 3d Tiles loader which allows to use 3D Tiles inside your own instance of ThreeJS: https://github.com/nytimes/three-loader-3dtiles , I think it's a good example of what we would want, it allows to use that 3d tiles inside react-three-fiber : https://github.com/nytimes/three-loader-3dtiles/tree/dev/examples/r3f , and it manages tiles loading etc thanks to a callback telling it about the current view frustum.

Best, Thank you for your time @gchoqueux !

**Comment by gchoqueux** [comment]
As part of the development of `webxr` support, 
We are planning to expose the itowns update process and that the itowns `objects` support a standalone use in `three.js`.

thank you also for the discussion
